### ISSUE TITLE
Can Fosscord be made LibreJS compilant?

### ISSUE THREAD

**Issue by LordRishav** [fosscord/fosscord-client#14]
GNU LibreJS is a Mozilla Firefox addon which filters out non-free and non-trivial JavaScript from any website. As Fosscord is already Free Software, licensed under the AGPL, can it be possible to make it LibreJS compliant? The GNU project has also published some guides on how to free your Javascript [here](https://www.gnu.org/software/librejs/free-your-javascript.html) and [here](https://www.gnu.org/software/librejs/manual/html_node/Setting-Your-JavaScript-Free.html). Thank you.

**Comment by JacobKotrla** [comment]
Even if all the code is public, I think it's important to let the users know that it is and where they can find it. For example how would a user find the corresponding source code to this script?: https://app.fosscord.com/cdn-cgi/challenge-platform/h/b/orchestrate/jsch/v1

I assume it is in this repository somewhere, but I cannot seem to find it. If someone visited that website without first finding this repository, they would see nothing to suggest that the application the website sends them is free software. I don't think LibreJS-compliance is strictly necessary, as long as it's easy for a human to find the corresponding source code (and know that it's corresponding) without first executing the script itself.

**Comment by TheArcaneBrony** [comment]
@LordRishav due to how its built, this isn't reasonable. We have no control over what other node packages do with their licensing.
LibreJS is a curse to any serious attempt at web development, as using any framework at all will make it automatically not be LibreJS compliant. This only works if you're writing everything from scratch which would be a huge undertaking, way bigger then what we do now.

**Comment by ghost** [comment]
> I assume it is in this repository somewhere, but I cannot seem to find it. If someone visited that website without first finding this repository, they would see nothing to suggest that the application the website sends them is free software. I don't think LibreJS-compliance is strictly necessary, as long as it's easy for a human to find the corresponding source code (and know that it's corresponding) without first executing the script itself.

Yeah if LibreJS isn't feasible, this way is good as well. Have the link of the source code hosted on Github inside a source file.
Having an extension automatically check whether a file is free or not is convenient still, but I can understand why it is not possible.

**Comment by TheArcaneBrony** [comment]
Well, the issue lies in dependencies. As soon as you import anything its automatically not librejs compliant, which will result in your code not executing since a script it loads isnt compliant. Though, yes, we can add github links to our code.
